Sources/NostrSDK/Events/RelayListMetadataEvent.swift

```diff
@@ -0,0 +1,154 @@
+    /// If the marker is omitted, the relay is used for both read and write.
+    ///
+    /// A `nil` value is returned if the relay URL string is invalid or the marker is invalid.
+    ///
+    /// > Note: [NIP 65 - Relay List Metadata](https://github.com/nostr-protocol/nips/blob/master/65.md)
+    public init?(tag: Tag) {
+        guard tag.name == "r", let relayURL = try? RelayURLValidator.shared.validateRelayURLString(tag.value) else {
+            return nil
+        }
+
+        switch tag.otherParameters.first {
+        case "read":
+            marker = .read
+        case "write":
+            marker = .write
+        case .none:
+            marker = .readAndWrite
+        case .some:
+            return nil
+        }
+
+        self.relayURL = relayURL
+    }
+
+    /// Creates a ``UserRelayMetadata`` from a relay ``URL`` and ``Marker``.
+    ///
+    /// A `nil` value is returned if the relay URL string is invalid.
+    ///
+    /// > Note: [NIP 65 - Relay List Metadata](https://github.com/nostr-protocol/nips/blob/master/65.md)
+    public init?(relayURL: URL, marker: Marker) {
+        guard let validatedRelayURL = try? RelayURLValidator.shared.validateRelayURL(relayURL) else {
+            return nil
+        }
+        self.relayURL = validatedRelayURL
+        self.marker = marker
+    }
+
+    /// The ``Tag`` that represents the user relay metadata that can be used in a ``RelayListMetadataEvent``.
+    public var tag: Tag {
+        let otherParameters: [String]
+        switch marker {
+        case .read:
+            otherParameters = ["read"]
+        case .write:
+            otherParameters = ["write"]
+        case .readAndWrite:
+            otherParameters = []
+        }
+        return Tag(name: "r", value: relayURL.absoluteString, otherParameters: otherParameters)
+    }
+}
+
+public extension EventCreating {
+    /// Creates a ``RelayListMetadataEvent`` (kind 10002).
+    /// - Parameters:
+    ///   - relayMetadataList: The list of ``UserRelayMetadata``.
+    ///   - keypair: The ``Keypair`` to sign the event with.
+    func relayListMetadataEvent(withRelayMetadataList relayMetadataList: [UserRelayMetadata], signedBy keypair: Keypair) throws -> RelayListMetadataEvent {
+        // Using an ordered dictionary to retain the order of the list while de-duplicating the data.
+        var deduplicatedMetadata = OrderedDictionary<URL, UserRelayMetadata>()
+
+        for metadata in relayMetadataList {
+            if let existingMetadata = deduplicatedMetadata[metadata.relayURL] {
+                // If the user relay metadata is identical between the duplicates,
+                // or if the existing one already has a read and write marker, skip it.
+                guard existingMetadata.marker != metadata.marker && existingMetadata.marker != .readAndWrite else {
+                    continue
+                }
```

Comment:
I would recommend using `if` rather than `guard` here. `guard` is better for protecting the last line in a function from all the undesirable cases. This is difficult to read here. And since you have a comment above it that uses the word `if`, it would be easier to check if the code matches the comment if you use `if`.

```suggestion
                if existingMetadata.marker == metadata.marker || existingMetadata.marker == .readAndWrite {
                    continue
                }
```